pkg/controller/nodelink/nodelink_controller.go

```diff
@@ -234,10 +246,43 @@ func (r *ReconcileNodeLink) Reconcile(request reconcile.Request) (reconcile.Resu
 		}
 	}
 
-	// TODO(alberto): bring nodeRef logic here and drop cluster-api node controller
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileNodeLink) updateNodeRef(machine *mapiv1beta1.Machine, node *corev1.Node) error {
```

Comment:
Could we add a comment here, or at the call site, about the reason for "touching" the machine resource on node status change. It's not immediately clear.

Reply:
done https://github.com/openshift/machine-api-operator/pull/336/files#diff-69885f1e6e1f6b4cb889ce80021c95a1R273